### PR TITLE
Handling Concurrent calls to the setup method

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/RudderClient.ts [hunk 1]
```diff
@@ -92,10 +96,14 @@ async function setup(writeKey: string, configuration: Configuration = {}, option
...
-  const config = await configure(writeKey, configuration);
-  logDebug("setup: created config");
-  await bridge.setup(config,options);
-  logDebug("setup: setup completed");
+  // Acquire a lock before calling the setup of Native Modules
+  lock.acquire("lock", async function(done) {
+    const config = await configure(writeKey, configuration);
+    logDebug("setup: created config");
+    await bridge.setup(config,options);
+    logDebug(`setup: setup completed of RudderSDK version ${SDK_VERSION}`);
```
Comment (SumanthPuram) [inline src/RudderClient.ts:104]: Shall we move this out. If setup gets blocked, we won't get to know the version. 

We should log this very early in setup.

**Review by SumanthPuram — DISMISSED** [review]

File: src/RudderClient.ts (hunk 1 above)
Comment (desusai7) [reply src/RudderClient.ts:104]: Yes, will move it

File: src/Logger.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ const WARN = 2;
 const ERROR = 1;
 const NONE = 0;
 
-let _logLevel = ERROR;
+let _logLevel = DEBUG;
```
Comment (arnab-p) [inline src/Logger.ts:8]: We may want to revert this

**Review by arnab-p — APPROVED** [review]

File: src/Logger.ts (hunk 2 above)
Comment (desusai7) [reply src/Logger.ts]: Reverted it and moved the initialization of the logger to the top of setup

**Review by arnab-p — APPROVED** [review]

**Pull request merged**